src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -81466,6 +81469,527 @@ public U Test<U>(U val)
+{
+    public MyAttribute(string s, string s2 = ""foo"") { }
+}
+
+[MyAttribute(null)] //1
+class C {} 
+
+[MyAttribute(null, null)] // 2, 3
+class D {}
+
+[MyAttribute(null, ""foo"")] // 4
+class E {}
+
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (7,14): warning CS8625: Cannot convert null literal to non-nullable reference or unconstrained type parameter.
+                // [MyAttribute(null)] //1
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(7, 14),
+                // (10,14): warning CS8625: Cannot convert null literal to non-nullable reference or unconstrained type parameter.
+                // [MyAttribute(null, null)] // 2, 3
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(10, 14),
+                // (10,20): warning CS8625: Cannot convert null literal to non-nullable reference or unconstrained type parameter.
+                // [MyAttribute(null, null)] // 2, 3
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(10, 20),
+                // (13,14): warning CS8625: Cannot convert null literal to non-nullable reference or unconstrained type parameter.
+                // [MyAttribute(null, "foo")] // 4
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(13, 14)
+                );
+        }
+
+        [Fact]
+        public void AttributeArgument_Constructor_NullLiteral_WithNamedArguments()
+        {
+            var source =
+@"#nullable enable
+class MyAttribute : System.Attribute
+{
+    public MyAttribute(string s, string s2 = ""foo"", string? s3 = ""bar"") { }
+}
+
+[MyAttribute(""foo"", s2: null, s3: null)] //1
+class C {}
+
+[MyAttribute(s3: null, s2: null, s: ""foo"")] // 2
+class D {}
+
+[MyAttribute(s3: null, s2: ""bar"", s: ""foo"")]
+class E {}
+
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (7,25): warning CS8625: Cannot convert null literal to non-nullable reference or unconstrained type parameter.
+                // [MyAttribute("foo", s2: null, s3: null)] //1
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(7, 25),
+                // (10,28): warning CS8625: Cannot convert null literal to non-nullable reference or unconstrained type parameter.
+                // [MyAttribute(s3: null, s2: null, s: "foo")] // 2
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(10, 28)
+                );
+        }
+
+        [Fact]
+        public void AttributeArgument_Constructor_NullLiteral_DisabledEnabled()
+        {
+            var source =
+@"
+#nullable enable
+class MyAttribute : System.Attribute
+{
+    public MyAttribute(string s, string s2) { }
+}
+
+[MyAttribute(null, //1
+#nullable disable
+null
+#nullable enable
+)] 
+class C {}
+
+[MyAttribute(
+#nullable disable
```

Comment:
What if I do this in the middle of a named parameter? For example:
```C#
using System;
#nullable enable
public class C : Attribute
{
    public C(string test) { }
}

[C(
    test:
    #nullable disable
    null
    #nullable enable
    )]
class D { }
```

Reply:
Added extra cases to cover

---
In reply to: [262704796](https://github.com/dotnet/roslyn/pull/33816#discussion_r262704796) [](ancestors = 262704796)